src/utils.ts

```diff
@@ -43,3 +46,44 @@ export function mergeLists<T>(list1: ReadonlyArray<T>, list2: ReadonlyArray<T>):
 
   return result
 }
+
+function objectKeys(object: object): Set<string> {
+  return new Set(Object.keys(object))
+}
+
+function areSetsEqual<T>(set1: Set<T>, set2: Set<T>): boolean {
+  const difference = new Set(set1)
+  for (const elem of set2) {
+    if (difference.has(elem)) {
+      difference.delete(elem)
+    } else {
+      // FIXME: uncommenting this causes change in behavior, where spreadInsert()
```

Comment:
How about adding an (optional?) additional parameter to the function, like `allowExcessiveProps`?

Reply:
hm do you view this as intentional behavior? As in, do you intend for
`spreadInsert({ a: 1 }, { a: 2, b: 1 })` to be valid? If not, I don't think we should allow it in the API and consider this a bug to be fixed later.

If you do intend for this behavior, then we can either add the parameter like you suggested, or we could change the name of this function, like maybe `isSubsetOf`.

Reply:
tldr; Let's not add the parameter and `return false` here, even if it's a breaking change.

Fair point about the name of the function… I am not 100% sure yet about allowing or preventing excessive properties, but we should probably prevent it.

My first gut feeling was rather in favor of allowing excessive props, so you can pass an object that is a superset of the thing you want to insert to `spreadInsert()` and it would still work.

On second thought… this is probably not gonna be possible, anyway, as `spreadInsert()` doesn't know the schema at runtime, so it would not be able to tell which props are fields of the record to insert and which ones are excessive.

Reply:
Sounds good to me. Is it alright if we do that in the next PR? I'd prefer this PR not change behavior at all.

Reply:
Of course!